### PR TITLE
reef: cephadm: use importlib.metadata for querying ceph_iscsi's version

### PR DESCRIPTION
**Pull request by zdover23**
use importlib.metadata for querying ceph_iscsi's version and fallback to pkg_resources. as the former is only available in Python 3.8, while the latter is deprecated.

Refs https://tracker.ceph.com/issues/66201

This commit is functionally equivalent to a Reef-targeted backport of https://github.com/ceph/ceph/pull/57685.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/cephadm/cephadm.py [hunk 1]
```diff
@@ -1035,17 +1035,32 @@ def get_container_binds():
...
+                    '/usr/bin/python3',
+                    '-c',
+                    s,
+                ],
+                verbosity=CallVerbosity.QUIET,
+            )
+
+        out, _, code = python(
+            "from importlib.metadata import version; print(version(`ceph_iscsi`))"
+
```
Comment (jdurgin) [inline src/cephadm/cephadm.py:1054]: need a ) here

File: src/cephadm/cephadm.py [hunk 2]
```diff
@@ -1035,17 +1035,32 @@ def get_container_binds():
...
-    def get_version(ctx, container_id):
-        # type: (CephadmContext, str) -> Optional[str]
-        version = None
-        out, err, code = call(ctx,
-                              [ctx.container_engine.path, 'exec', container_id,
-                               '/usr/bin/python3', '-c',
-                               "import pkg_resources; print(pkg_resources.require('ceph_iscsi')[0].version)"],
-                              verbosity=CallVerbosity.QUIET)
+      def python(s: str) -> Tuple[str, str, int]:
+            return call(
```
Comment (jdurgin) [inline src/cephadm/cephadm.py:1039]: can keep the # type comment

File: src/cephadm/cephadm.py [hunk 3]
```diff
@@ -1035,17 +1035,32 @@ def get_container_binds():
...
     @staticmethod
-    def get_version(ctx, container_id):
-        # type: (CephadmContext, str) -> Optional[str]
-        version = None
-        out, err, code = call(ctx,
-                              [ctx.container_engine.path, 'exec', container_id,
-                               '/usr/bin/python3', '-c',
-                               "import pkg_resources; print(pkg_resources.require('ceph_iscsi')[0].version)"],
-                              verbosity=CallVerbosity.QUIET)
+      def python(s: str) -> Tuple[str, str, int]:
```
Comment (jdurgin) [inline src/cephadm/cephadm.py:1038]: no need to change the function name/signature

Comment (jdurgin) [reply src/cephadm/cephadm.py:1038]: this python() function should be nested inside the get_version() function, similar to main

File: src/cephadm/cephadm.py (hunk 1 above)
Comment (zdover23) [reply src/cephadm/cephadm.py:1056]: I have made this change.

File: src/cephadm/cephadm.py (hunk 2 above)
Comment (zdover23) [reply src/cephadm/cephadm.py:1040]: I believe that I have made this change correctly.

File: src/cephadm/cephadm.py (hunk 3 above)
Comment (zdover23) [reply src/cephadm/cephadm.py:1039]: Here, I'm out of my depth and need to be taught to read.

**Review by adk3798 — APPROVED** [review]
From what I can tell recent pushes have just been changing the formatting, so as long as make check passes this LGTM

**Comment by zdover23** [comment]
> From what I can tell recent pushes have just been changing the formatting, so as long as make check passes this LGTM

Thanks, @adk3798. I'm just learning (at the feet of the extremely patient Josh) about how very finicky Python can be in connection to whitespace. I hope that all my... uh... mucking about... hasn't been too much of an inconvenience. Thanks for your forbearance.

**Comment by zdover23** [comment]
jenkins retest this please

**Comment by zdover23** [comment]
jenkins test make check

**Comment by zdover23** [comment]
jenkins test make check

**Pull request merged**